### PR TITLE
Add --no-diff option (fixes #2514)

### LINKED ISSUE #2514: Option to disable diff
Hi,

We currently have a use case where we're testing a large array for deep equality against something returned from a function. The problem is that when the test fails the output in the reporter is very long and clogs all the tests.

Is there any option to disable the diff? In this case it would be very useful, since we don't really need to know where the output is different from the expected result, but just if it's deeply equal to it or not.

Thanks!

### PR DESCRIPTION
**Pull request by CapacitorSet**
Adds a `--no-diff` flag, which disables diffs completely (takes priority over `--inline-diffs`). Made to fix #2514.

### CODE REVIEW

File: mocha.js [hunk 1]
```diff
@@ -1641,6 +1641,20 @@ Mocha.prototype.useInlineDiffs = function (inlineDiffs) {
 };
 
 /**
+ * Do not show diffs at all.
```
Comment (boneskull) [inline mocha.js]: please don't commit changes to this file.  it's lame we keep it under VCS (b/c bower)

**Comment by CapacitorSet** [comment]
Actually, nevermind, it appears that `err`s already have a `showDiff` property. I'll use that instead.

**Comment by CapacitorSet** [comment]
Done! I squashed the changes as recommended in CONTRIBUTING.md.

**Comment by CapacitorSet** [comment]
Ping @boneskull. This PR still has the *pr-needs-work* label, but I think it's fixed now.

**Comment by rhali786** [comment]
Hey did this get pulled out? I don't see it in the code. I was looking forward to this.

**Comment by boneskull** [comment]
someone needs to review this and probably resolve the conflicts.  as to how soon that'll happen, I can't tell you

**Comment by boneskull** [comment]
closed by 7ceaaf22c934afcd3b39aa57d15c4324f28c1e27 and 575d83d76b97dfa426ee9f1aa436e27f9e96b283

**Pull request closed without merging**